src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -526,10 +546,14 @@ public void setDatasetBatchSize(Integer datasetBatchSize) {
     this.datasetBatchSize = datasetBatchSize;
   }
 
-  public Dataset updateDatasetIndex(Integer datasetId, Integer userId, Instant indexDate)
-      throws SQLException {
-    datasetServiceDAO.updateDatasetIndex(datasetId, userId, indexDate);
-    return datasetDAO.findDatasetById(datasetId);
+  private void synchronizeDatasetInESIndex(Dataset dataset, User user) {
```

Comment:
Doesn't this need to check `updatedDataset.getIndexedDate() != null`?

Reply:
This is a good catch. There are two usages in this class, one is intended to update the index because we've already updated the index date and the second set of usages should be checking the date, but they are not - I need to refactor this.